protocol/contracts/core/VaultProxy.sol

```diff
@@ -113,23 +71,49 @@ contract VaultProxy is Ownable {
     }
 
     /**
-    * Toggle whether faith is claimable
+    * @notice Transfer out amount of token to provided address
     */
-    function toggleFaithClaimEnabled() external onlyOwner {
-        faithClaimEnabled = !faithClaimEnabled;
-    }
-
-    /**
-    * transfer out amount of token to provided address
-    */
-    function withdraw(address token, address to, uint256 amount) external onlyOwner {
-        require(to != address(0), "to address zero");
+    function withdraw(address token, address to, uint256 amount) public onlyOwner {
+        if (to == address(0)) {
+            revert SendToAddressZero();
+        }
 
         if (token == address(0)) {
             (bool sent,) = payable(to).call{value: amount}("");
-            require(sent, "send failed");
+            if (!sent) {
+                revert WithdrawSendFailed();
+            }
         } else {
             SafeERC20.safeTransfer(IERC20(token), to, amount);
         }
     }
+
+    /**
+     * Allows users to withdraw their Temple from a vault before the exit window, forfeits any yield
+     */
+    function exitVaultEarly(uint256 amount, Vault vaultErc) external {
+        SafeERC20.safeTransferFrom(IERC20(vaultErc), msg.sender, address(this), amount);
+        uint256 shareAmount = vaultErc.toSharesAmount(amount);
+        uint256 templeBal = vaultErc.toTokenAmount(shareAmount);
+        SafeERC20.safeTransfer(temple, msg.sender, templeBal);
```

Comment:
Would love to have an event here if possible, with something like:
`emit EarlyWithdraw(msg.sender, templeBal, vaultErc)`
`event EarlyWithdraw(address account, uint256 amount, address vault)`